### PR TITLE
[FEATURE] Allowing `schema`  to be passed in as `batch_spec_passthrough` in Spark

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
Changes proposed in this pull request:
- Follow-up to #5897
- Schemas in Spark Dataframes are defined as `StructType`, which is not serializable.
- This PR adds the `schema.jsonValue()` call which translates the object into a `json`, whenever a schema is passed in as a `batch_spec_passthrough` parameter
- `schema` can be passed in at the `Asset`-level and `DataConnector`-level. 
- `schema` can also be passed in as part of  `BatchRequest` that is part of a `Checkpoint`. 
- The PR adds the necessary changes to `prepare_dump` and `convert_to_json_serializable()` and adds integration tests to `test_serialization.py`. 
- Also includes changes from #5917
### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/data_context/types/base.py [hunk 1]
```diff
@@ -273,6 +278,27 @@ class Meta:
     def validate_schema(self, data, **kwargs) -> None:  # type: ignore[no-untyped-def]
         pass
 
+    @pre_dump
+    def prepare_dump(self, data, **kwargs):
```
Comment (Shinnnyshinshin) [inline great_expectations/data_context/types/base.py:282]: `batch_spec_passthrough` can be at the `Asset`-level

File: great_expectations/data_context/types/base.py [hunk 2]
```diff
@@ -735,6 +761,27 @@ def validate_schema(self, data, **kwargs):
     def make_data_connector_config(self, data, **kwargs):
         return DataConnectorConfig(**data)
 
+    @pre_dump
+    def prepare_dump(self, data, **kwargs):
```
Comment (Shinnnyshinshin) [inline great_expectations/data_context/types/base.py:765]: `batch_spec_passthrough` can be at the `DataConnector`-level

File: tests/core/test_serialization.py [hunk 3]
```diff
@@ -701,133 +710,555 @@ def test_checkpoint_config_and_nested_objects_are_serialized(
...
-                    "data_connector_name": "my_data_connector",
-                    "data_asset_name": "users",
-                    "data_connector_query": {"partition_index": -1},
-                    "batch_spec_passthrough": {"reader_options": {"header": True}},
-                },
-                id="06871341-f028-4f1f-b8e8-a559ab9f62e1",
+@pytest.mark.parametrize(
+    "checkpoint_config,expected_serialized_checkpoint_config",
+    [
+        pytest.param(
```
Comment (Shinnnyshinshin) [inline tests/core/test_serialization.py:916]: the same tests are now parameterized :)

File: tests/core/test_serialization.py [hunk 4]
```diff
@@ -701,133 +710,555 @@ def test_checkpoint_config_and_nested_objects_are_serialized(
...
     )
 
 
-@pytest.mark.integration
-def test_serialization_of_datasource_with_nested_objects_spark(spark_session):
-    datasource_config: DatasourceConfig = DatasourceConfig(
-        name="taxi_data",
-        class_name="Datasource",
-        module_name="great_expectations.datasource",
-        execution_engine=ExecutionEngineConfig(
```
Comment (Shinnnyshinshin) [inline tests/core/test_serialization.py:778]: the same tests are now parameterized :)

**Review by anthonyburdi — APPROVED** [review]
Nice work! LGTM!

File: great_expectations/data_context/types/base.py [hunk 5]
```diff
@@ -735,6 +761,27 @@ def validate_schema(self, data, **kwargs):
...
+
+        batch_spec_passthrough_config = data.get("batch_spec_passthrough")
+        if batch_spec_passthrough_config:
+            reader_options: dict = batch_spec_passthrough_config.get("reader_options")
+            if reader_options:
+                schema = reader_options.get("schema")
+                if schema and isinstance(schema, StructType):
+                    data["batch_spec_passthrough"]["reader_options"][
+                        "schema"
+                    ] = schema.jsonValue()
```
Comment (NathanFarmer) [inline great_expectations/data_context/types/base.py:782]: ❤️

Comment (NathanFarmer) [inline great_expectations/data_context/types/base.py:782]: ❤️

File: tests/core/test_serialization.py [hunk 6]
```diff
@@ -701,133 +710,555 @@ def test_checkpoint_config_and_nested_objects_are_serialized(
...
+                                        "header": True,
+                                        "schema": StructType(
+                                            [
+                                                StructField(
+                                                    "a", IntegerType(), True, None
+                                                ),
+                                                StructField(
+                                                    "b", IntegerType(), True, None
+                                                ),
+                                            ]
```
Comment (NathanFarmer) [inline tests/core/test_serialization.py]: This is leveling up huge!!

**Review by NathanFarmer — APPROVED** [review]
Love it!

**Pull request merged**